### PR TITLE
Refactoring and simplifying CQA table, updating icons

### PR DESCRIPTION
**Pull request by hannahbergam**
This pr covers three updates. I did them together because they're all small and cover the same area. 

1. Makes sure that clicking 'decide later' clears out the selected course offering (which makes the version and unit dropdowns disappear)- a fix!
2. Updates the icons to be right unless selected, then pointing down if selected to match advanced settings
3. Updates the marketing audience selectors to collapse the table if pressed again
4. (3A) Refactors the icon and onClick functions to be shared to avoid repeating code

New icon status:
Screenshot 2023-04-12 at 1 25 54 PM

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/TEACH-414
- https://codedotorg.atlassian.net/browse/TEACH-411

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ export default function CurriculumQuickAssign({updateSection, sectionCourse}) {
     updateSection('course', {});
     if (marketingAudience !== '') {
       setMarketingAudience('');
+      setSelectedCourseOffering(null);
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:47]: This is (1) from PR description

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssignTopRow.jsx [hunk 2]
```diff
@@ -17,6 +17,21 @@ export default function CurriculumQuickAssignTopRow({
   marketingAudience,
   updateMarketingAudience
 }) {
+  // Determines which icon to render for each audience
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssignTopRow.jsx:20]: This is (3A)

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssignTopRow.jsx [hunk 3]
```diff
@@ -17,6 +17,21 @@ export default function CurriculumQuickAssignTopRow({
...
+  const marketingAudienceIcon = audience => {
+    return marketingAudience === audience ? 'caret-down' : 'caret-right';
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssignTopRow.jsx:22]: This is (2)

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssignTopRow.jsx [hunk 4]
```diff
@@ -17,6 +17,21 @@ export default function CurriculumQuickAssignTopRow({
...
+  };
+
+  // If the given audience is already selected, deselect it.
+  // Otherwise, set to this audience
+  const determineMarketingAudience = newAudience => {
+    if (newAudience === marketingAudience) {
+      updateMarketingAudience('');
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssignTopRow.jsx:29]: This is (3)

**Review by rshipp — APPROVED** [review]
lgtm!

**Pull request merged**